### PR TITLE
refactor: migrate pokedex to jina 5

### PR DESCRIPTION
**Pull request by JamesTang-jinaai**
Resolved: #188

### CODE REVIEW

File: pokedex-with-bit/pods/doc.yml [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
-  index_filename: doc.gzip
+!CompoundIndexer
+components:
+  - !NumpyIndexer
+    with:
+      index_filename: vec.gz
+    metas:
+      name: vecidx  # a customized name
+      workspace: $WORKDIR
+  - !DataURIPbIndexer
```
Comment (JoanFM) [inline pokedex-with-bit/pods/doc.yml:9]: I think the problem comes from this, if it works alone it would have the proper Drivers attached, but working in a CompoundIndexer it may not, you can take inspiration on CrossModal Search, in the ImageIndexer there is the usage of `BinaryPbIndexer` that uses as one of its Drivers a Buffer2URI {}

File: pokedex-with-bit/flow-index.yml [hunk 2]
```diff
@@ -10,13 +10,16 @@ pods:
     parallel: $PARALLEL
     timeout_ready: 600000
     read_only: true
-  chunk_idx:
-    uses: pods/chunk.yml
+  vec_idx:
+    uses: pods/vec.yml
     shards: $SHARDS
-    separated_workspace: true
+    separated_workspace: false
```
Comment (JoanFM) [inline pokedex-with-bit/flow-index.yml:16]:
```suggestion
    separated_workspace: true
```

File: pokedex-with-bit/flow-index.yml [hunk 3]
```diff
@@ -10,13 +10,16 @@ pods:
...
   doc_idx:
     uses: pods/doc.yml
-    needs: gateway
+    shards: $SHARDS
+    separated_workspace: false
```
Comment (JoanFM) [inline pokedex-with-bit/flow-index.yml:20]:
```suggestion
    separated_workspace: true
```

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by JoanFM** [comment]
Need to set logserver to False in `flow-index`.

But I do not know why `docidx` does not seem to receive any message from the gateway? Although u can try with images from the real pokemon examples that I did not find

**Comment by JoanFM** [comment]
@JamesTang-jinaai I did a push to ur branch, pull locally and test please

File: pokedex-with-bit/pods/doc.yml [hunk 4]
```diff
@@ -1,6 +1,23 @@
...
+  workspace: $WORKDIR
+requests:
+  on:
+    ControlRequest:
+      - !ControlReqDriver {}
+    SearchRequest:
+      - !KVSearchDriver
+        with:
+          granularity_range: [0, 0]
+          adjacency_range: [0, 1]
```
Comment (nan-wang) [inline pokedex-with-bit/pods/doc.yml:15]:
```suggestion
          - !ExcludeQL
            with:
              fields:
                - embedding
```

File: pokedex-with-bit/app.py [hunk 5]
```diff
@@ -5,14 +5,14 @@
 import sys
 
 from jina.flow import Flow
-
-num_docs = os.environ.get('MAX_DOCS', 50000)
-image_src = 'data/**/*.png'
+# num_docs = os.environ.get('MAX_DOCS', 50000)
+num_docs = os.environ.get('MAX_DOCS', 5)
+image_src = 'data/*.png'
```
Comment (nan-wang) [inline pokedex-with-bit/app.py:10]: revert

File: pokedex-with-bit/app.py [hunk 6]
```diff
@@ -5,14 +5,14 @@
 import sys
 
 from jina.flow import Flow
-
-num_docs = os.environ.get('MAX_DOCS', 50000)
-image_src = 'data/**/*.png'
+# num_docs = os.environ.get('MAX_DOCS', 50000)
```
Comment (nan-wang) [inline pokedex-with-bit/app.py:8]: revert

**Review by nan-wang — CHANGES_REQUESTED** [review]

**Comment by JamesTang-jinaai** [comment]
Now the docker file woks.

File: pokedex-with-bit/app.py [hunk 7]
```diff
@@ -29,7 +28,7 @@ def index():
     f = Flow.load_config('flow-index.yml')
 
     with f:
-        f.index_files(image_src, batch_size=64, read_mode='rb', size=num_docs)
+        f.index_files(image_src, batch_size=5, read_mode='rb', size=num_docs, skip_dry_run=True)
```
Comment (JoanFM) [inline pokedex-with-bit/app.py:31]: revert these changes

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: pokedex-with-bit/pods/doc.yml (hunk 4 above)
Comment (JoanFM) [reply pokedex-with-bit/pods/doc.yml:15]: is it removing or adding on top of it? I think we need to keep the `adjacency_range`

File: pokedex-with-bit/pods/doc.yml [hunk 8]
```diff
@@ -1,6 +1,32 @@
...
+      - !ExcludeQL
+        with:
+          granularity_range: [0, 2]
+          adjacency_range: [0, 1]
+          fields:
+            - embedding
+    IndexRequest:
+      - !LogInfoDriver {}
+      - !Buffer2URI {}
+      - !LogInfoDriver {}
```
Comment (JoanFM) [inline pokedex-with-bit/pods/doc.yml:25]: remember to removed all these `LogInfoDriver`

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Review by nan-wang — APPROVED** [review]
LGTM👍

File: pokedex-with-bit/pods/doc.yml (hunk 8 above)
Comment (nan-wang) [reply pokedex-with-bit/pods/doc.yml]: @JamesTang-jinaai Would you please remove this?
